.gitmodules

```diff
@@ -10,3 +10,6 @@
 [submodule "third_party/date"]
 	path = third_party/date
 	url = https://github.com/HowardHinnant/date.git
+[submodule "third_party/googletest"]
```

Comment:
Another way to build `gtest` is to download the source at configure time and build it (as shown in the [gtest docs](https://github.com/google/googletest/blob/master/googletest/README.md)). This has the advantage that its fully integrated into your parent project's build system as opposed to relying on the user checking out the submodule before building. Since we're already using a number of submodules, I guess this isn't a real concern here?

Reply:
I much prefer having all dependencies tracked by git rather than pulling down things dynamically at build time.

Reply:
The "download" option also uses `git` to get the source (you can specify branch/commit/tag etc to specify a commit). Its actually very similar to what submodule does, but follows the strictly out-of-source-tree build philosophy, where the build step doesn't touch the source tree at all, and removing build artifacts is simply a matter of deleting the "build" dir.

If we weren't using submodules at all, I would've preferred this approach (I've used it past projects before without issues).